microshift.sh

```diff
@@ -0,0 +1,120 @@
+
+function configure_host {
+    sudo dnf install -y git osbuild-composer composer-cli ostree rpm-ostree \
+      cockpit-composer cockpit-machines bash-completion podman genisoimage \
+      createrepo yum-utils selinux-policy-devel jq wget lorax
+    sudo systemctl enable osbuild-composer.socket --now
+    sudo systemctl enable cockpit.socket --now
+    sudo firewall-cmd --add-service=cockpit --permanent
+    sudo chmod 0755 $HOME
+}
+
+function enable_repo {
+    # Add respective repo to fetch the microshift and dependency packages
+    sudo subscription-manager repos \
+       --enable rhocp-4.12-for-rhel-8-$(uname -i)-rpms \
+       --enable fast-datapath-for-rhel-8-$(uname -i)-rpms
+}
+
+function download_microshift_rpm {
+    local pkgDir=$(mktemp -p /tmp -d tmp-rpmXXX)
+    sudo dnf download --downloadonly --downloaddir ${pkgDir} microshift microshift-networking microshift-selinux >/dev/null 2>&1
+    sudo chmod 0777 ${pkgDir}
+    echo ${pkgDir}
+}
+
+function create_iso {
+    local pkgDir=$1
+    rm -fr microshift
+    # Clone microshift repo
+    git clone -b release-4.12 https://github.com/openshift/microshift.git
+    cp podman_changes.ks microshift/
+    pushd microshift
+    sudo chmod 0755 $HOME
+    ssh-keygen -t ecdsa -b 521 -N "" -f id_ecdsa_crc -C "core"
+    sed -i '/# customizations/,$d' scripts/image-builder/config/blueprint_v0.0.1.toml
+    sed -i 's/redhat/core/g' scripts/image-builder/config/kickstart.ks.template
+    sed -i '/--bootproto=dhcp/a\network  --hostname=api.crc.testing' scripts/image-builder/config/kickstart.ks.template
+    sed -i '$e cat podman_changes.ks' scripts/image-builder/config/kickstart.ks.template
+    scripts/image-builder/cleanup.sh -full
+    scripts/image-builder/build.sh -microshift_rpms ${pkgDir} -pull_secret_file ${OPENSHIFT_PULL_SECRET_PATH} -authorized_keys_file $(realpath ../id_ecdsa_crc.pub)
+    popd
+    rm -fr $pkgDir
+}
+
+enable_repo
+configure_host
+microshift_pkg_dir=$(download_microshift_rpm)
+create_iso ${microshift_pkg_dir}
+sudo cp microshift/_output/image-builder/microshift-installer-*.iso /var/lib/libvirt/images/microshift-installer.iso
+
+OPENSHIFT_RELEASE_VERSION=$(sudo dnf list | awk "/^microshift\./ {print \$2;}")
+create_json_description ${BUNDLE_TYPE}
+
+# For microshift we create a empty kubeconfig file
+# to have it as part of bundle
+mkdir -p ${INSTALL_DIR}/auth
+touch ${INSTALL_DIR}/auth/kubeconfig
+
+# Start the VM with generated ISO
+sudo virt-install \
+    --name ${CRC_VM_NAME} \
+    --vcpus 2 \
+    --memory 2048 \
+    --arch=${ARCH} \
+    --disk path=/var/lib/libvirt/images/${CRC_VM_NAME}.qcow2,size=31 \
+    --network network=default,model=virtio \
+    --nographics \
+    --cdrom /var/lib/libvirt/images/microshift-installer.iso \
+    --events on_reboot=restart \
+    --autoconsole none \
+    --wait
+
+# Sleep for 3 mins because after starting the vm, kickstart file takes time to
+# install from iso
+sleep 180
```

Comment:
Can't this be replaced with `virt-install ... --wait 3`?

Reply:
Done use `-wait 5` to be safer side.